invenio/modules/knowledge/api.py

```diff
@@ -662,17 +668,10 @@ def get_kbd_values_for_bibedit(tag, collection="", searchwith="",
                        if present, '%' is substituted with /searcwith/.
                        If absent, /searchwith/ is searched for in /tag/.
     """
-    dkbname = "tmp_dynamic_"+tag+'_'+expression
-    kb_id = add_kb(kb_name=dkbname, kb_type='dynamic')
-    # get the kb name since it may be catenated by a number
-    # in case there are concurrent calls.
-    kb_name = get_kb_name(kb_id)
-    add_kb_mapping(kb_name, tag, expression, collection)
-    # now, get stuff
-    myvalues = get_kbd_values(kb_name, searchwith)
-    # the tmp dyn kb is now useless, delete it
-    delete_kb(kb_name)
-    return myvalues
+    collection = Collection.query.filter_by(name=collection).first()
+    kbdefs = models.KnwKBDDEF(output_tag=tag, expression=expression,
```

Comment:
I'm not sure if this is safe. You are creating SQLAlchemy object that it's binded to a session and if it runs in request than it's auto-commited to db. Can you change `get_kbd_values_by_kbdefs` to accept `dict` with values from `kbdefs.to_dict()` instead of `KnwKBDDEF` instance? Then it should be fine to do `s/models.KnwKBDDEF/dict/`.